Zend/tests/grammar/regression_004.phpt

```diff
@@ -9,7 +9,10 @@ class Obj
     function return(){} // valid
 }
 
-function echo(){} // not valid
+function echo(){print("test\n");} // valid
+\echo();
```

Comment:
this seems quite confusing to me: https://3v4l.org/ksZdk

Reply:
Yeah, I think we should not allow declaring functions for keywords that can be used in a function-like way. Basically, if you have `function kw() {}` and writing `kw();` causes a compile error, that's okay. However, if you have `function kw() {}` and writing `kw();` compiles but will do something other than calling the function `kw()`, then I think we should prevent the declaration.

Reply:
That seems easier to remember than using a cut-off date. For global functions, that would be:


  ~~INCLUDE~~ | ~~INCLUDE_ONCE~~ | ~~EVAL~~ | ~~REQUIRE~~ | ~~REQUIRE_ONCE~~ | OR | XOR | AND
| INSTANCEOF | ~~NEW~~ `new ('expression')` is allowed | ~~CLONE~~ | ~~EXIT~~ | ~~IF~~ | ~~ELSEIF~~ | ELSE | ENDIF | ~~ECHO~~ | DO | ~~WHILE~~ | ENDWHILE
| ~~FOR~~ | ENDFOR | ~~FOREACH~~ | ENDFOREACH | ~~DECLARE~~ | ENDDECLARE | AS | TRY | CATCH (requires {}) | FINALLY
| ~~THROW~~ | ~~USE~~ | INSTEADOF | GLOBAL | VAR | ~~UNSET~~ | ~~ISSET~~ | ~~EMPTY~~ | ~~CONTINUE~~ (continue (2);) | GOTO
| ~~FUNCTION~~ | CONST | ~~RETURN~~ | ~~PRINT~~ | ~~YIELD~~ | ~~LIST~~ | ~~SWITCH~~ | ENDSWITCH | CASE | DEFAULT | ~~BREAK~~ (break (2);)
| ~~ARRAY~~ | CALLABLE | EXTENDS | IMPLEMENTS | NAMESPACE | TRAIT | INTERFACE | CLASS
| __CLASS__ | __TRAIT__ | __FUNCTION__ | __METHOD__ | __LINE__ | __FILE__ | __DIR__ | __NAMESPACE__
| ~~FN~~ (`\fn() => expr,` confusing in array key) | MATCH (requires {}) | ENUM

// semi_reserved:
| STATIC | ABSTRACT | FINAL | PRIVATE | PROTECTED | PUBLIC | READONLY (also see other PR)